src/components/Header/index.ts

```diff
@@ -1,4 +1,14 @@
+import { connect } from 'react-redux';
 import { withRouter } from 'react-router-dom';
+import { compose } from 'redux';
+import { leaveRoom } from 'store/room/actions';
 import Header from './Header';
 
-export default withRouter(Header);
+const mapDispatchToProps = {
+  leaveRoom,
+};
+
+export default compose(
```

Comment:
try this one should work the same way `connect(null, mapDispatchToProps)(withRouter(Header))`